### PR TITLE
Add case-insensitive routing

### PR DESCRIPTION
**Pull request by tdotclare**
Allows configuring case-insensitive routing (#2354).

```swift
// Enables case-insensitive routing.
// Defaults to false.
app.routes.caseInsensitive = true
```

### CODE REVIEW

**Comment by tdotclare** [comment]
Someone asked yesterday on Discord about case insensitive routing. I don't know that I'd ever actually ADVISE someone to do this, but since TrieRouter can do it, this PR adds a configurable setting to `app.routes` so DefaultResponder can grab it and pass it to TrieRouter.

Whether this is a *Good Idea* is up for debate and feel free to skip approving it if not, hah

**Comment by 0xTim** [comment]
I'm +1 for at least exposing this

File: Sources/Vapor/Routing/Routes.swift [hunk 1]
```diff
@@ -3,6 +3,7 @@ public final class Routes: RoutesBuilder, CustomStringConvertible {
     
     /// Default value used by `HTTPBodyStreamStrategy.collect` when `maxSize` is `nil`.
     public var defaultMaxBodySize: ByteCount
+    public var caseInsenstive: Bool
```
Comment (tanner0101) [inline Sources/Vapor/Routing/Routes.swift:6]: Doc block here, please.

**Comment by tanner0101** [comment]
Please format the PR title / body as release notes, too.

**Comment by tdotclare** [comment]
Initial PR comments reproduced:

* Adds Application.Routes.caseInsensitive flag to allow configuring case insensitive routes prior to boot (defaults to false)
* DefaultResponder will configure TrieRouter appropriately if set to true
* Appropriate test

### Allows configuring Vapor to use case insensitive route handling ###
EG:
```
app.routes.caseInsensitive = true
app.routes.get("foo") {...}
app.run()
...
myserver.com/foo -> "200 OK"
myserver.com/FOO -> "200 OK"
```

Note that case-insensitive routing is somewhat less performant, and only applies to constant PathComponents - eg: `app.routes.get("foo", ":id")` maintains case of URL request to parameter `id`, so `myserver.com/FoO/uSeRiD` would correctly hit the registered route handler but within the route handler, the value of `id` will still be `uSeRiD`.

File: Sources/Vapor/Routing/Routes.swift (hunk 1 above)
Comment (tdotclare) [reply Sources/Vapor/Routing/Routes.swift:8]: Done

**Comment by tanner0101** [comment]
These changes are now available in [4.12.0](https://github.com/vapor/vapor/releases/tag/4.12.0)

**Comment by maclee99** [comment]
missing an "I" in the app.routes.caseInsenstive

**Pull request merged**